src/components/pages/UserProfile.js

```diff
@@ -34,53 +33,49 @@ const useStyles = () => ({
     top: 0,
   },
   box: {},
-});
+}));
 
-class UserProfile extends React.Component {
-  async componentDidMount() {
-    const { location, currentUser, loadUserDetails, history } = this.props;
-    const userId = location.pathname.split("=")[1];
-    let response = await loadUserDetails({
-      userId,
-      currentUserId: currentUser ? currentUser._id : null,
-    });
-    if (response !== "success") {
-      history.push("/UserNotFound");
-    }
-  }
+const UserProfile = () => {
+  const history = useHistory();
+  const location = useLocation();
+  const userDetail = useSelector((state) => state.userDetail);
+  const currentUser = useSelector((state) => state.currentUser.user);
+  const dispatch = useDispatch();
+  const classes = useStyles();
 
-  render() {
-    let { userDetail, currentUser, classes, openReviewModal } = this.props;
+  useEffect(() => {
+    async function loadUser() {
+      try {
+        const userId = location.pathname.split("=")[1];
+        let response = await dispatch(
+          loadUserDetails({
+            userId,
+            currentUserId: currentUser ? currentUser._id : null,
+          })
+        );
+        if (response !== "success") {
+          history.push("/UserNotFound");
+        }
+      } catch (error) {
+        console.log("Failed to load User Profile page");
```

Comment:
Might not be a good idea to use console.logs in production code for our submission. Better to use snackbar to displayError instead.

Reply:
the error isn't meant for frontend to display. It's rather a help for developers to see what's happening ;)